tap_hubspot/__init__.py

```diff
@@ -888,24 +888,44 @@ def sync_engagements(STATE, ctx):
 
 
 def sync_deal_pipelines(STATE, ctx):
+    bookmark_key = "updatedAt"
+    start = utils.strptime_with_tz(get_start(STATE, "deal_pipelines", bookmark_key))
+    max_bk_value = start
     catalog = ctx.get_catalog_from_id(singer.get_currently_syncing(STATE))
     mdata = metadata.to_map(catalog.get("metadata"))
     schema = catalog["schema"]
     singer.write_schema(
         "deal_pipelines", schema, ["pipelineId"], catalog.get("stream_alias")
     )
-    LOGGER.info("sync_deal_pipelines")
+    LOGGER.info(f"sync deal_pipelines from {start}")
+
     data = request(get_url("deal_pipelines")).json()
     with Transformer(UNIX_MILLISECONDS_INTEGER_DATETIME_PARSING) as bumble_bee:
-        for row in data:
+        for row in data["results"]:
+            modified_time = None
             row = replace_na_with_none(row)
-            record = bumble_bee.transform(row, schema, mdata)
-            singer.write_record(
-                "deal_pipelines",
-                record,
-                catalog.get("stream_alias"),
-                time_extracted=utils.now(),
+            if bookmark_key in row:
+                timestamp_millis = row[bookmark_key]
+            elif "createdAt" in row:
+                # Hubspot returns timestamps in millis
+                timestamp_millis = row["createdAt"]
+            modified_time = datetime.datetime.fromtimestamp(
+                timestamp_millis / 1000.0, datetime.timezone.utc
             )
+            if modified_time and modified_time >= max_bk_value:
+                max_bk_value = modified_time
+
+            if not modified_time or modified_time > start:
```

Comment:
maybe invert this if sentence and add the comments describing what this check is guarding against. It will be hard to grok when you get back here in a few months (do double check that inverted if logic):
```python
if modified_time and modified_time <= start:
    ...
```

Reply:
do you mean I should do
```
if modified_time and modified_time <= start:
    continue
```
?

Reply:
https://images.app.goo.gl/Vt77DurjEeieCD2r9

Reply:
But then I need to change every where in the code. Can I do it next time when I DRY the code?[Please](https://giphy.com/gifs/kitten-please-pretty-Q94xQWspTUkShljj8P)